app/uk/gov/hmrc/gform/instructions/PageFieldConverter.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.instructions
+
+import play.api.i18n.Messages
+import uk.gov.hmrc.gform.controllers.AuthCacheWithForm
+import uk.gov.hmrc.gform.eval.smartstring.SmartStringEvaluator
+import uk.gov.hmrc.gform.fileupload.Envelope
+import uk.gov.hmrc.gform.instructions.FormModelInstructionSummaryConverter.PageField
+import uk.gov.hmrc.gform.sharedmodel.LangADT
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ ComponentType, FormComponent, SectionNumber }
+import uk.gov.hmrc.gform.validation.ValidationResult
+
+trait PageFieldConverter[T <: ComponentType] {
```

Comment:
Is this trait really necessary?

As I see is that it forces you to do unsafe `asInstanceOf` cast, and pass almost a dozen of arguments to the functions which do not need them. 

Why don't use `IsText` and other extractors to stay consistent with the rest of the code base?

Reply:
The idea was to have separate implementations of summary converters for each scenario - i.e InstructionPDF now, summary and summary PDF later. Also adding a new converter for a new type doesnt touch the main code, just an additional instance required. However, I do agree the long list of arguments and the casting is a bit annoying

Reply:
@josef-vlach I have refactored the code. Please take a look